app/views/modules/player/_video_js_element.html.erb

```diff
@@ -0,0 +1,235 @@
+                             }
+                           else
+                             {
+                               "children": [
+                                 'playToggle',
+                                 'volumePanel',
+                                 'progressControl',
+                                 'currentTimeDisplay',
+                                 'timeDivider',
+                                 'durationDisplay'
+                               ]
+                             }
+                           end %>
+
+  <% @videojs_options = {"aspectRatio": section_info[:is_video] ? '16:9' : '1:0',
+                         "autoplay": false,
+                         "width": @player_width || 480,
+                         "height": @player_height || 270,
+                         "bigPlayButton": section_info[:is_video] ? true : false,
+                         "poster": section_info[:is_video] ?  section_info[:poster_image] : false,
+                         "fluid": true,
+                         "responsive": true,
+                         "preload": "auto",
+                         "controlBar": control_bar_options,
+                         "userActions": {
+                           hotkeys: true
+                         }
+                       }.compact.to_json %>
+
+  <div class='video-container' style="width:100%; height:100%;">
+    <video-js id="video-js-embed-<%= @master_file.id %>"
+            class="vjs-big-play-centered"
+            controls
+            style="width:100%; height:100%;"
+            data-setup='<%= @videojs_options %>'>
+      <% section_info[:stream_hls].each do |hls| %>
+        <source src="<%= hls[:url] %>" type="application/x-mpegURL" data-quality="<%= hls[:quality] %>" label="<%= hls[:quality] %>"/>
+      <% end %>
+      <% if section_info[:caption_paths].present? %>
+        <% section_info[:caption_paths].each do |c| %>
+          <track <% if c[:label] %>label="<%= c[:label] %>" <% end %> srclang="<%= c[:language] %>" kind="subtitles" type="<%= c[:mime_type] %>" src="<%= c[:path] %>"></track>
+        <% end %>
+      <% end %>
+    </video-js>
+    <% if section_info[:is_video] %>
+      <label class='video-title' id='video-title-<%= @master_file.id %>'>
+        <a class='video-link' id='video-link-<%= @master_file.id %>' href="<%= master_file_url(@master_file.id) %>" target="_blank" rel="noreferrer noopener"><%= @master_file.title || @master_file.media_object.title %></a>
+      </label>
+    <% end %>
+  </div>
+<% end %>
+
+<% # These script tags are for pulling in videojs and the quality selector plug-in from their CDNs.
+   # May be worth considering pulling them into our local asset pipeline. %>
+<script src="https://vjs.zencdn.net/7.10.2/video.min.js"></script>
+<script src="https://unpkg.com/@silvermine/videojs-quality-selector/dist/js/silvermine-videojs-quality-selector.min.js"></script>
```

Comment:
Similarly, to load these through asset pipeline we could add the following to the `application.js`;
```
//= require video.js/dist/video.min.js
//= require @silvermine/videojs-quality-selector/dist/js/silvermine-videojs-quality-selector.min.js
```